modin/engines/base/io/file_dispatcher.py

```diff
@@ -116,24 +131,67 @@ def file_size(cls, f):
         f.seek(cur_pos, os.SEEK_SET)
         return size
 
+    @staticmethod
+    def _s3_path(s3_path: str, glob: bool) -> List[str]:
+        """
+        Get s3 file paths.
+
+        Parameters
+        ----------
+        s3_path: str
+            String of s3 path.
+        glob: bool
+            True if the path is a glob path.
+
+        Returns
+        -------
+        list[str]
+            List of s3 files based on the path.
+        """
+        # S3FS does not allow captial S in s3 addresses.
+        if s3_path[0] == "S":
+            s3_path = "{}{}".format("s", s3_path[1:])
+
+        import s3fs as S3FS
+        from botocore.exceptions import NoCredentialsError
+
+        def get_file_path(fs_handle) -> List[str]:
+            if not glob:
+                if fs_handle.exists(s3_path):
+                    return [s3_path]
+                else:
+                    return []
+            else:
+                s3_paths = fs_handle.glob(s3_path)
+                s3_addresses = ["{}{}".format("s3://", path) for path in s3_paths]
+                return s3_addresses
+
+        s3fs = S3FS.S3FileSystem(anon=False)
+        try:
+            return get_file_path(s3fs)
+        except NoCredentialsError:
+            pass
+        s3fs = S3FS.S3FileSystem(anon=True)
+        return get_file_path(s3fs)
+
     @classmethod
-    def file_exists(cls, file_path):
+    def file_exists(cls, file_path: str) -> bool:
```

Comment:
Instead of modifying the original, can we override it? This should simplify the code for the existing readers.

Reply:
Yes, I think I was thinking along these lines but didnt go as far. I have since made these changes